### DISCUSSION TITLE
How can I call Java methods from existing Maven libraries (and their dependencies)?

### DISCUSSION THREAD

**Discussion by axkr** [nextjournal/clerk#542] [Q&A]
How can I call Java methods from existing Maven libraries (and their dependencies)?

Is there an example, how to use a JSR 223 Scripting language from within clerk?

**Comment by jackrusher** [comment]
These seem like general Clojure questions rather than Clerk-specific ones. I recommend you have a look at the Clojure documentation. :)

**Reply by axkr** [reply]
Thank you, I will look there again.

But the idea I have in mind, will be a "meta description of dependencies" similar to JBang 
- https://www.jbang.dev/documentation/guide/latest/index.html

which loads Java libraries from Maven Central (or local repository)

These libraries are "prepared with a clerk/clojure interface definition" (for example in a manifest file definition in the JAR), which can use the output formats from clerk by running a "script command" defined in the library.
There are so many Java libraries which could easily be reused in such an environment.

As an example JSR223 uses such a `javax.script.ScriptEngineFactory` Factory definition for the scripting:
![image](https://github.com/nextjournal/clerk/assets/3109793/94217486-2b4a-41ef-a9af-4383918bf323)

**Comment by jackrusher** [comment]
There is a mechanism for loading libraries from Maven, local jars, git repos, &c, in Clojure already:

https://clojure.org/guides/deps_and_cli

**Reply by axkr** [reply]
Yes, how about the missing part of calling a JSR223 script engine via `eval()`method?

Example:
```
ScriptEngine#eval("Plot(Sin(x),{x,-10,10})");
```

and creating clerk plotting output formats?
